shodan/client.py

```diff
@@ -384,24 +384,24 @@ def _request(self, function, params, service='shodan', method='get', json_data=N
                     # Otherwise lets raise the error message
                     error = u'{}'.format(e)
 
-            raise APIError(error)
+            raise APIError(error, data.status_code)
         elif data.status_code == 403:
-            raise APIError('Access denied (403 Forbidden)')
+            raise APIError('Access denied (403 Forbidden)', data.status_code)
         elif data.status_code == 502:
-            raise APIError('Bad Gateway (502)')
+            raise APIError('Bad Gateway (502)', data.status_code)
 
         # Parse the text into JSON
         try:
-            data = data.json()
+            parsed_data = data.json()
         except ValueError:
-            raise APIError('Unable to parse JSON response')
+            raise APIError('Unable to parse JSON response', data.status_code)
 
         # Raise an exception if an error occurred
-        if type(data) == dict and 'error' in data:
-            raise APIError(data['error'])
+        if type(parsed_data) == dict and 'error' in parsed_data:
```

Comment:
Would it make sense to use `isinstance()` rather than `type()` to check for dict? Either way would work, but I think that using `isinstance()` [would be preferred](https://stackoverflow.com/questions/25231989/how-to-check-if-a-variable-is-a-dictionary-in-python)

Reply:
Yeah I guess it would make more sense, I was just using the old code and added my new logic to it but I get what you are saying and I changed it!

Reply:
Looks good!